src/Umbraco.Infrastructure/Runtime/CoreInitialComposer.cs

```diff
@@ -177,6 +196,170 @@ public override void Compose(Composition composition)
+                .Append<SettingsSection>()
+                .Append<PackagesSection>()
+                .Append<UsersSection>()
+                .Append<MembersSection>()
+                .Append<FormsSection>()
+                .Append<TranslationSection>();
+
+            // register known content apps
+            composition.ContentApps()
+                .Append<ListViewContentAppFactory>()
+                .Append<ContentEditorContentAppFactory>()
+                .Append<ContentInfoContentAppFactory>();
+
+            // register published router
+            composition.RegisterUnique<IPublishedRouter, PublishedRouter>();
+
+            // register *all* checks, except those marked [HideFromTypeFinder] of course
+            composition.HealthChecks()
+                .Add(() => composition.TypeLoader.GetTypes<HealthCheck>());
+
+
+            composition.WithCollectionBuilder<HealthCheckNotificationMethodCollectionBuilder>()
+                .Add(() => composition.TypeLoader.GetTypes<IHealthCheckNotificationMethod>());
+
+            composition.RegisterUnique<IContentLastChanceFinder, ContentFinderByConfigured404>();
+
+            composition.ContentFinders()
+                // all built-in finders in the correct order,
+                // devs can then modify this list on application startup
+                .Append<ContentFinderByPageIdQuery>()
+                .Append<ContentFinderByUrl>()
+                .Append<ContentFinderByIdPath>()
+                //.Append<ContentFinderByUrlAndTemplate>() // disabled, this is an odd finder
+                .Append<ContentFinderByUrlAlias>()
+                .Append<ContentFinderByRedirectUrl>();
+
+            composition.Register<UmbracoTreeSearcher>(Lifetime.Request);
+
+            composition.SearchableTrees()
+                .Add(() => composition.TypeLoader.GetTypes<ISearchableTree>());
+
+            // replace some services
+            composition.RegisterUnique<IEventMessagesFactory, DefaultEventMessagesFactory>();
+            composition.RegisterUnique<IEventMessagesAccessor, HybridEventMessagesAccessor>();
+            composition.RegisterUnique<ITreeService, TreeService>();
+            composition.RegisterUnique<ISectionService, SectionService>();
+
+            composition.RegisterUnique<IExamineManager, ExamineManager>();
+
+            // register distributed cache
+            composition.RegisterUnique(f => new DistributedCache(f.GetInstance<IServerMessenger>(), f.GetInstance<CacheRefresherCollection>()));
+
+
+            composition.Register<ITagQuery, TagQuery>(Lifetime.Request);
+
+            composition.RegisterUnique<HtmlLocalLinkParser>();
+            composition.RegisterUnique<HtmlUrlParser>();
+            composition.RegisterUnique<HtmlImageSourceParser>();
+            composition.RegisterUnique<RichTextEditorPastedImages>();
+
+            // we should stop injecting UmbracoContext and always inject IUmbracoContextAccessor, however at the moment
```

Comment:
Is this comment still relevant? Thought this was already taken care of?

Reply:
It only seemed to be nessasary due to `RenderNoContentController`, but I fixed that up